draft-ietf-tls-tls13.md

```diff
@@ -3592,104 +3589,23 @@ ClientHello.client_version. No single value will guarantee interoperability
-  (3).
-
-session_id_length
-: This field MUST have a value of zero for a client that claims to
-  support TLS 1.2.
-
-challenge_length
-: The length in bytes of the client's challenge to the server to
-  authenticate itself.  Historically, permissible values are between
-  16 and 32 bytes inclusive.  When using the SSLv2 backward-
-  compatible handshake the client SHOULD use a 32-byte challenge.
-
-cipher_specs
-: This is a list of all CipherSpecs the client is willing and able
-  to use.  In addition to the 2.0 cipher specs defined in {{SSL2}},
-  this includes the TLS cipher suites normally sent in
-  ClientHello.cipher_suites, with each cipher suite prefixed by a
-  zero byte.  For example, the TLS cipher suite {0x00,0x0A} would be
-  sent as {0x00,0x00,0x0A}.
-
-session_id
-: This field MUST be empty.
-
-challenge
-: Corresponds to ClientHello.random.  If the challenge length is
-  less than 32, the TLS server will pad the data with leading (note:
-  not trailing) zero bytes to make it 32 bytes long.
-{:br }
-
-Note: Requests to resume a TLS session MUST use a TLS client hello.
+## Compatibility with SSL
 
-## Avoiding Man-in-the-Middle Version Rollback
+The security of SSL 2.0 {{SSL2}} is considered insufficient for the reasons enumerated
+in [RFC6176].
 
-When TLS clients fall back to Version 2.0 compatibility mode, they MUST use
-special PKCS#1 block formatting. This is done so that TLS servers will reject
-Version 2.0 sessions with TLS-capable clients.
+Implementations MUST NOT send an SSL version 2.0 compatible CLIENT-HELLO.
+Implementations MUST NOT negotiate TLS 1.3 or later using an SSL version 2.0 compatible
+CLIENT-HELLO. Implementations MAY accept an SSL version 2.0 compatible CLIENT-HELLO in
+order to negotiate older versions of TLS, however this is not recommended.
+Implementations MUST NOT send or accept any records with a version less than { 3, 0 }.
 
-When a client negotiates SSL 2.0 but also supports TLS, it MUST set the
-right-hand (least-significant) 8 random bytes of the PKCS padding (not
-including the terminal null of the padding) for the RSA encryption of the
-ENCRYPTED-KEY-DATA field of the CLIENT-MASTER-KEY to 0x03 (the other padding
-bytes are random).
+The security of SSL 3.0 {{SSL3}} is considered insufficient for the reasons enumerated
+in [I-D.ietf-tls-sslv3-diediedie].
 
-When a TLS-capable server negotiates SSL 2.0 it SHOULD, after decrypting the
-ENCRYPTED-KEY-DATA field, check that these 8 padding bytes are 0x03. If they
-are not, the server SHOULD generate a random value for SECRET-KEY-DATA, and
-continue the handshake (which will eventually fail since the keys will not
-match). Note that reporting the error situation to the client could make the
-server vulnerable to attacks described in {{BLEI}}.
+Implementations MUST NOT send a ClientHello or ServerHello with the protocol version set
+to { 3, 0 } or less. Any endpoint receiving a Hello message with the protocol version set
```

Comment:
I think this would be clearer if it read ClientHello.version or ServerHello.version, since then it would be double clear it's not the record layer.